File: codeWars/katas/breaking_chocolate_problem/breaking_chocolate_problem.ts
// Breaking chocolate problem
export {};

function breakChocolate(n: number, m: number): number {
  if (n <= 0 || m <= 0) {
    return 0; // Invalid input, no breaks needed
  }

  return n * m - 1; // Minimum number of breaks needed
}

console.log(breakChocolate(2, 1)); // Output: 1
console.log(breakChocolate(3, 1)); // Output: 2
console.log(breakChocolate(0, 5)); // Output: 0 (Invalid input)
console.log(breakChocolate(4, 4)); // Output: 15

File: codeWars/katas/breaking_chocolate_problem/breaking_chocolate_problem.py
# Breaking chocolate problem


def breakChocolate(n, m):
    if n <= 0 or m <= 0:
        return 0  # Invalid input, no breaks needed

    return n * m - 1  # Minimum number of breaks needed


print(breakChocolate(2, 1))  # Output: 1
print(breakChocolate(3, 1))  # Output: 2
print(breakChocolate(0, 5))  # Output: 0 (Invalid input)
print(breakChocolate(4, 4))  # Output: 15
